### PR TITLE
1691 initialize on batch create

### PR DESCRIPTION
**Pull request by rwmcintosh**
Cloning the simulation prevents changes to the original reference from affecting the batch once created

### CODE REVIEW

File: src/OSPSuite.Core/Domain/ModelCoreSimulation.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@ public interface IModelCoreSimulation : IObjectBase, IWithCreationMetaData, IWit
       /// <summary>
       ///    Build configuration used to create the simulation. May be null
       /// </summary>
-      IBuildConfiguration BuildConfiguration { get; }
+      IBuildConfiguration BuildConfiguration { get; set; }
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/ModelCoreSimulation.cs:14]: This may be a breaking change in pksim and Mobi FYI

File: src/OSPSuite.R/Domain/ConcurrentRunSimulationBatch.cs [hunk 2]
```diff
@@ -23,9 +23,9 @@ public class ConcurrentRunSimulationBatch : IDisposable, IWithId
 
       internal int MissingBatchesCount => Math.Max(0, SimulationBatchRunValues.Count - _simulationBatches.Count);
 
-      public ConcurrentRunSimulationBatch(IModelCoreSimulation simulation, SimulationBatchOptions simulationBatchOptions)
+      public ConcurrentRunSimulationBatch(Simulation simulation, SimulationBatchOptions simulationBatchOptions)
```
Comment (msevestre) [inline src/OSPSuite.R/Domain/ConcurrentRunSimulationBatch.cs:26]: Why this change here?

**Review by msevestre — CHANGES_REQUESTED** [review]
I am not sure that changing the public interface to simulation is necessary. Explain?

File: src/OSPSuite.R/Domain/ConcurrentRunSimulationBatch.cs (hunk 2 above)
Comment (rwmcintosh) [reply src/OSPSuite.R/Domain/ConcurrentRunSimulationBatch.cs:26]: As far as I can tell, this is always called with R.Domain.Simulation, which isn't set up for cloning. In another spot, I need to access the property CoreSimulation to actually create a clone of it. So rather than cast, I think it's ok that we just use the object.

File: src/OSPSuite.Core/Domain/ModelCoreSimulation.cs (hunk 1 above)
Comment (rwmcintosh) [reply src/OSPSuite.Core/Domain/ModelCoreSimulation.cs:14]: Meaning, we'll have to implement the setters once this is merged?

File: src/OSPSuite.R/Domain/ConcurrentRunSimulationBatch.cs (hunk 2 above)
Comment (rwmcintosh) [reply src/OSPSuite.R/Domain/ConcurrentRunSimulationBatch.cs:26]: maybe I just need a default constructor - hold on

Comment (rwmcintosh) [reply src/OSPSuite.R/Domain/ConcurrentRunSimulationBatch.cs]: Yes, with a default constructor for R.Domain.Simulation the clone manager is happy.

Comment (msevestre) [reply src/OSPSuite.R/Domain/ConcurrentRunSimulationBatch.cs]: What about you don't create a clone then but just instantiate a new Simulation?? If there was not default constructor, there was probalbgy a reason

Comment (msevestre) [reply src/OSPSuite.R/Domain/ConcurrentRunSimulationBatch.cs]: so return new Simulation(_clonedModelCoreSimulation)

Comment (rwmcintosh) [reply src/OSPSuite.R/Domain/ConcurrentRunSimulationBatch.cs]: If this goes back to IModelCoreSimulation, then there's no time during the batch creation where I can get access to the underlying CoreSimulation without a cast. I need either the default constructor and clone, or cast to R.Domain.Simulation to get access to the underlying CoreSimulation so I can clone it and then instantiate an R.Domain.Simulation.

**Comment by msevestre** [comment]
Merged. In order for the behavior to be available in R, we will need to update the dll
